packages/codemod/src/upgrade-legacy.ts

```diff
@@ -49,37 +49,58 @@ const upgradeLegacy = async () => {
       const program = getCollectionFromSource(blitzConfigFile)
       const parsedProgram = program.get()
 
-      // Clear file
-      parsedProgram.value.program.body = []
-
-      // We create an object property eg. {withBlitz: withBlitz}
       let withBlitz = j.objectProperty(j.identifier("withBlitz"), j.identifier("withBlitz"))
-      // Then set the shorthand to true so we get {withBlitz}
       withBlitz.shorthand = true
-
-      /* Declare the variable using the object above that equals to a require expression, eg.
-          const {withBlitz} = require("@blitzjs/next")
-        */
-      let blitzDeclare = j.variableDeclaration("const", [
-        j.variableDeclarator(
-          j.objectPattern([withBlitz]),
-          j.callExpression(j.identifier("require"), [j.stringLiteral("@blitzjs/next")]),
+      let config = program.find(j.AssignmentExpression, {
+        left: {
+          type: "MemberExpression",
+          object: {
+            type: "Identifier",
+            name: "module",
+          },
+          property: {
+            type: "Identifier",
+            name: "exports",
+          },
+        },
+      })
+      let createdConfig = config.get().value.right
+      const middlewareArray = program.find(j.Identifier, (node) => node.name === "middleware")
+      if (middlewareArray.size() > 0) {
+        if (createdConfig.properties) {
+          createdConfig.properties = createdConfig.properties.filter(
+            (node) => node.key.name !== "middleware",
+          )
+        } else {
+          if (createdConfig.arguments[0].properties)
+            createdConfig.arguments[0].properties = createdConfig.arguments[0].properties.filter(
+              (node) => node.key.name !== "middleware",
+            )
+          else {
+            throw new ExpectedError(
+              "Middleware has not been removed from next.config.js. It has to be removed manually",
+            )
+          }
+        }
+      }
```

Comment:
I think we should remove this. Then in the `Move middleware to blitz server file` step we can handle the middlewares.

Reply:
Would you be able to remove this block? Then I'll merge the PR and then I'll create a new one fixing the middlewares in the correct step.

Reply:
@dillonraphael I have now committed the requested change